src/main/java/de/gebit/plugins/autoconfig/handlers/sonarlint/SonarLintHandler.java

```diff
@@ -0,0 +1,72 @@
+package de.gebit.plugins.autoconfig.handlers.sonarlint;
+
+import com.intellij.openapi.module.Module;
+import com.intellij.openapi.module.ModuleManager;
+import com.intellij.openapi.project.Project;
+import de.gebit.plugins.autoconfig.UpdateHandler;
+import de.gebit.plugins.autoconfig.handlers.AbstractHandler;
+import de.gebit.plugins.autoconfig.model.Binding;
+import de.gebit.plugins.autoconfig.model.SonarLintConfiguration;
+import org.jetbrains.annotations.NonNls;
+import org.sonarlint.intellij.config.Settings;
+import org.sonarlint.intellij.config.module.SonarLintModuleSettings;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+/**
+ * Configuration handler used to synchronise SonarLint module bindings in projects.
+ */
+public class SonarLintHandler extends AbstractHandler implements UpdateHandler<SonarLintConfiguration> {
+	private static final @NonNls String CONFIG_SCHEMA_JSON = "/schema/sonarlint.schema.json";
+
+	public static final @NonNls String CONFIG_FILE_NAME = "autoconfigSonarLint.yaml";
+
+	@Override
+	public String getFileName() {
+		return CONFIG_FILE_NAME;
+	}
+
+	@Override
+	public String getJsonSchema() {
+		return CONFIG_SCHEMA_JSON;
+	}
+
+	@Override
+	public String getUpdaterName() {
+		return "SonarLint configuration updater";
+	}
+
+	@Override
+	public Class<SonarLintConfiguration> getConfigurationClass() {
+		return SonarLintConfiguration.class;
+	}
+
+	@Override
+	public List<String> updateConfiguration(SonarLintConfiguration configuration, Project project) {
+		if (configuration.getBindings() != null) {
+			Map<String, Module> moduleMap = Arrays.stream(ModuleManager.getInstance(project).getModules())
+					.collect(Collectors.toMap(Module::getName, m -> m));
+			// create a dummy list for the purpose of collecting the number of changes only
+			List<String> changedModuleConfigs = new ArrayList<>();
+			for (Binding binding : configuration.getBindings()) {
+				moduleMap.entrySet().stream().filter(e -> e.getKey().matches(binding.getModulePattern())).forEach(e -> {
```

Comment:
`moduleMap.entrySet().stream().filter(moduleEntry -> modukeEntry.getKey().matches(binding.getModulePattern())).forEach(moduleEntry -> {`

Reply:
Descriptive lambda names are better here because not every editor displays type inferences.

Reply:
This piece of code is not longer necessary because of the use of the module handler extension point.